### ISSUE TITLE
Support Trackpoint Scrolling

### ISSUE THREAD

**Issue by RoiEXLab** [triplea-game/triplea#1443]
From #760 

> I would also like to add, that movement is also canceled when I'm scrolling via trackpoint.
> 
> Like this:
> I press and hold the middle mouse button and now all my mouse movements are considered mouse wheel movements.
> 
> Normal (via touchpad) mouse wheel scrolling does not cancel the movement. So I guess the problem is the mouse button click, just like in the right click drag scrolling scenario.
> 
> So someone is working on this, maybe exclude besides right click also middle click from the cancel-movement-list.

@lensz We need more information about java keycodes etc. using your setup in order to implement this

**Comment by lensz** [comment]
Could you give me some hints on how I can give useful information on what happens exactly when pressing my mouse buttons/scrolling with my setup?

I presume it would be quite helpful to have information on what events the OS or triplea receives when I'm doing stuff with my mouse.

Is there a third party utility or some java code I could run to create some debug output?

By the way, I'm on linux.

**Comment by RoiEXLab** [comment]
[TestInputCodes.zip](https://github.com/triplea-game/triplea/files/697032/TestInputCodes.zip)
This zip file contains a jar file which will tell you input codes of your keyboard and mouse...
Just run it with java (openJDK or oracle JDK doesn't matter) and tell me what is shown when scrolling with your trackpad.

This tool could actually be useful for other test cases ^^

**Comment by lensz** [comment]
holding left click: mousePressed Button: 1 Modifiers: 16
    release left click: mouseClicked Button: 1 Modifiers: 16
    holding middle click: nothing
    release middle click: nothing
    quickly pressing middle click: mouseClicked Button: 2 Modifiers: 8
    holding right click: mousePressed Button: 3 Modifiers: 4
    release right click: mouseClicked Button: 3 Modifiers: 4

I wasn't able to generate a `mousePressed` event for the middle button.


Scrolling with the mousewheel (my trackpad) does not generate anything. 

----
Scrolling with the trackpoint:
holding middle mouse and scroll a bit: `mouseReleased: Button: 4/5 Modifier: 0`
The mouse Button is sometimes 4 and sometimes 5. It is kind of hard to get some sense into when its the one and when the other thing. Also: with the trackpoint it's kind of impossible to only scroll vertical or horizontal and that makes it even more difficult.

Let me know what's next :smiley:

**Comment by RoiEXLab** [comment]
[TestInputCodes.zip](https://github.com/triplea-game/triplea/files/697137/TestInputCodes.zip)
Updated the code to display mousewheel movement as well

> holding middle click: nothing
> release middle click: nothing

This is weird hmm, but at least it seems to be button 2.

> The mouse Button is sometimes 4 and sometimes 5. It is kind of hard to get some sense into when its the one and when the other thing. Also: with the trackpoint it's kind of impossible to only scroll vertical or horizontal and that makes it even more difficult.

Please try to figure out when it's button 4 and when it's button 5 - The thing I ask myself is, why scrolling with your trackpoint is even working. It should be impossible with your given information...

If you can, please try to record (using [OBS](https://obsproject.com/download) or whatever) what is happening on your end inside of triplea, since it's hard to follow what exactly you mean...

**Comment by lensz** [comment]
Could you maybe edit your jar another time and print all events additionally to stdout. This would help me.

**Comment by RoiEXLab** [comment]
Here you go:
[TestInputCodes.zip](https://github.com/triplea-game/triplea/files/697344/TestInputCodes.zip)

**Comment by lensz** [comment]
Nice. Now all events are clear and makes sense :smiley: 

Trackpoint scrolling:
scroll left: `mouseReleased: Button: 4 Modifier: 0`
scroll right `mouseReleased: Button: 5 Modifier: 0`
scroll up: `mouseWheelMoved: Rotation: -1.0 Modifier: 0`
scroll down: `mouseWheelMoved: Rotation: 1.0 Modifier: 0`

So interestingly enough, there is no middle mouse button event fired at all for scrolling, not at the beginning, not during and not at the end. That clears some things up I think.

**Comment by RoiEXLab** [comment]
Thanks!
I will look into the code tomorrow... 
As you probably know it's already 22:20 in Germany

**Comment by lensz** [comment]
I have to thank you for your dedication on the issue!

**Comment by RoiEXLab** [comment]
@lensz  #1451 should fix your issue

**Comment by lensz** [comment]
It works perfectly :)

I'm able to scroll vertical, horizontal and even diagonal with the trackpoint. And as icing on the cake it doesn't cancel the current move.

Now I can do this: https://gfycat.com/SeveralCrispAltiplanochinchillamouse

----
#1422 is now solved for me, too. (But I guess "normal" horizontal mouse wheel action still doesn't work)

**Comment by ron-murhammer** [comment]
@RoiEXLab @lensz Are we good to close this issue now?
